amplify/data/resource.ts

```diff
@@ -68,10 +69,20 @@ const schema = a
       .returns(a.ref("GenericFunctionResponse"))
       .authorization((allow) => [allow.guest(), allow.authenticated()])
       .handler(a.handler.function(AssignUsersToTeams)),
+    GenerateTeamCode: a
+      .mutation()
+      .arguments({
+        teamName: a.string().required(),
+        userId: a.string().required(),
```

Comment:
Instead of this could we add a boolean called `addCallerToTeam`. This way we can set it to true when a user creates a team and false when an admin is creating a team